pytorch_lightning/utilities/auto_restart.py

```diff
@@ -514,7 +512,7 @@ def patch_dataloader_iterator(
 def _add_capture_metadata_collate(dataloader: DataLoader) -> None:
     """Wrap default collate function to retrive captured dataset state dict when fault tolerant is enabled."""
     faut_tolerant_mode = _FaultTolerantMode.detect_current_mode()
-    if not faut_tolerant_mode.is_enabled:
+    if not faut_tolerant_mode.is_enabled or isinstance(dataloader.collate_fn, partial):
```

Comment:
is this to check if our collate is already attached? can you remind us in which case can it will occur? also, technically it would be possible that the user has a collate wrapped into partial :)

Reply:
Yes, this is done to prevent re-wrapping.

I added a better check

```py
(isinstance(dataloader.collate_fn, partial) and dataloader.collate_fn.func is _capture_metadata_collate)
```